### DISCUSSION TITLE
Can you guide me how to use "sort directory by date"?

### DISCUSSION THREAD

**Discussion by Max-Enrik** [CogentRedTester/mpv-file-browser#90] [General]
I need to sort directories by the date created. It would be nice to guide me so I can do it.

**Comment by CogentRedTester** [comment]
Download the script here: https://github.com/CogentRedTester/mpv-file-browser/issues/82#issuecomment-1342220863

Change the `.mtime` parts of the script to be `.ctime`.

Place the file into the `~~/script-modules/file-browser-addons/` folder as `csorting.lua`.

Make sure that [addons are enabled](https://github.com/CogentRedTester/mpv-file-browser#installation) in file-browser.

Run mpv, open file-browser and move to a directory. Press `^` and directories will be sorted by creation date. Press `^` again and they will go back to alphabetical order. To make date sorting enabled by default change `local do_sorting = false` near the top of the script to `local do_sorting = true`.

If things still do not work then run mpv with `mpv --log-file=log.txt --msg-level=file_browser='debug'`, attempt to do these steps, then upload the log file.

**Comment by tyf2018** [comment]
@CogentRedTester  After following the instructions, nothing happened. Can you please take a look? This is the log file.
[log.txt](https://github.com/user-attachments/files/15512718/log.txt)

Best regards.

**Reply by CogentRedTester** [reply]
It looks like your shell is including the `'` in the `debug`:

```
[   0.025][v][cplayer] Setting option 'msg-level' = 'file_browser='debug'' (flags = 8)
[   0.025][e][cplayer] Invalid message level ''debug''
[   0.025][e][cplayer] Error parsing option msg-level (option parameter could not be parsed)
[   0.025][f][cplayer] Setting commandline option --msg-level=file_browser='debug' failed.
[   0.025][i][cplayer] Exiting... (Fatal error)
```

You need to pass debug without the quotes to mpv, maybe ` mpv --log-file=log.txt --msg-level=file_browser=debug` would work?

**Comment by hunkyburrito** [comment]
The recent cache redesign (https://github.com/CogentRedTester/mpv-file-browser/commit/dd42418bd84a82207323b7f942147c0f3164e7f0) broke the sorting add-on (confirmed with a `git bisect`). It seems like the list is no longer getting sorted because it's re-using the cached views. 

<details>
<summary> Log: </summary>

```
[file_browser] imported parser sort-files from sort-files.lua
[file_browser] imported parser find from find.lua
[file_browser] registering directory alias ^dvd://.* /dev/dvd
[file_browser] registering directory alias ^bd://.* /dev/bd
[file_browser] registering directory alias ^cdda://.* /dev/cdrom
[file_browser] jumping to current directory
[file_browser] opening directory: /home/hunky/.config/mpv/
[file_browser] scanning files in "/home/hunky/.config/mpv/"
[file_browser] finding parser for "/home/hunky/.config/mpv/"
[file_browser] attempting parser: home-label
[file_browser] finding parser for "/home/hunky/.config/mpv/"
[file_browser] attempting parser: sort-files
[file_browser] attempting parser: ls
[file_browser] Starting subprocess: [ls, -1, -p, -A, -N, --zero, /home/hunky/.config/mpv/]
[file_browser] list returned from: ls
[file_browser] list returned from: home-label
[file_browser] Received custom keybind ^
[file_browser] opening directory: /home/hunky/.config/mpv/
[file_browser] found directory in cache
[file_browser] applying cache for /home/hunky/.config/mpv/
[file_browser] setting directory to /home/hunky/.config/mpv/
[file_browser] setting directory_label to ~/.config/mpv/
[file_browser] setting list to table: 0x76c4740ca990
[file_browser] setting selected to 1
[file_browser] setting selection to table: 0x76c4740cadc8
[file_browser] setting parser to table: 0x76c47410b638
[file_browser] setting empty_text to empty directory
[file_browser] setting co to thread: 0x76c4740ca8d8
[file_browser] Received custom keybind ^
[file_browser] opening directory: /home/hunky/.config/mpv/
[file_browser] found directory in cache
[file_browser] applying cache for /home/hunky/.config/mpv/
[file_browser] setting directory to /home/hunky/.config/mpv/
[file_browser] setting directory_label to ~/.config/mpv/
[file_browser] setting list to table: 0x76c4740ca990
[file_browser] setting selected to 1
[file_browser] setting selection to table: 0x76c4740cadc8
[file_browser] setting parser to table: 0x76c47410b638
[file_browser] setting empty_text to empty directory
[file_browser] setting co to thread: 0x76c4740ca8d8
```
</details>

Edit: Fixed it by adding `fb.clear_cache()` above `fb.rescan()`.

**Reply by CogentRedTester** [reply]
Should be fixed by now.
